src/main/scala/is/hail/expr/types/PhysicalType.scala

```diff
@@ -1,7 +1,329 @@
 package is.hail.expr.types
 
+import is.hail.annotations._
+import is.hail.asm4s._
+import is.hail.expr.ir.EmitMethodBuilder
+import is.hail.utils._
+
 abstract class PhysicalType {
   def virtualType: Type
+
+  def byteSize: Long
+
+  def alignment: Long
+
+  def unsafeOrdering(rightType: PhysicalType, missingGreatest: Boolean): UnsafeOrdering
+
+  def codeOrdering(mb: EmitMethodBuilder, other: PhysicalType): CodeOrdering
+}
+
+case class PDefault(virtualType: Type) extends PhysicalType {
```

Comment:
How are you going to use this?

Reply:
I want the current code to compile. I plan to remove PDefault.

Reply:
> I plan to remove PDefault.

OK, that's fine.